lib/msf/ui/console/command_dispatcher/db.rb

```diff
@@ -89,11 +85,21 @@ def active?
           end
 
           def cmd_set_data_service(service_id)
-            framework.db.set_data_service(service_id)
+            begin
+              framework.db.set_data_service(service_id)
+            rescue Exception => e
+              print_error "Unable to set data service: #{e.message}"
+            end
           end
 
           def cmd_list_data_services()
-            framework.db.print_data_services
+            framework.db.get_services_metadata.each {|metadata|
+              out = "id: #{metadata.id}, name: #{metadata.name}"
```

Comment:
Could consolidate status indicator into a single format string and reduce the block to: `print_line "id: #{metadata.id}, name: #{metadata.name}#{metadata.active ? " [active]" : nil}"`